simulation/validator/miner_data_handler.py

```diff
@@ -33,6 +33,10 @@ def save_responses(
             "num_simulations": simulation_input.num_simulations,
         }
 
+        if len(miner_predictions_data) == 0:
```

Comment:
can we move this call in the caller function here:
https://github.com/mode-network/synth-subnet/blob/main/simulation/validator/forward.py#L258
so the `save_responses` function is not called at all in case `miner_prediction_data` is empty?